homeassistant/components/sql/config_flow.py

```diff
@@ -82,8 +83,10 @@
 
 def validate_sql_select(value: str) -> str | None:
     """Validate that value is a SQL SELECT query."""
-    if not value.lstrip().lower().startswith("select"):
-        raise ValueError("Incorrect Query")
+    query_type = sqlparse.parse(value)[0].get_type().upper()
+    _LOGGER.debug("The SQL query is of type %s", query_type)
+    if not query_type == "SELECT":
+        raise ValueError("query_no_read_only")
```

Comment:
Make another exception here than `ValueError` to diff instead of using the text string below

Reply:
Added at https://github.com/home-assistant/core/pull/105921/commits/5eb739bac5456310d12a5d9d65b9855a5e193ca1